src/util/validate.js

```diff
@@ -10,50 +10,71 @@ import { readdir } from "node:fs/promises";
  * @throws {Error}                                     Throw error if options are invalid
  */
 export const validate = async (options, releaseInfo, version) => {
-  if (options.mode === "get") {
-    if (!releaseInfo.flavors.includes(options.flavor)) {
-      throw new Error(
-        `${options.flavor} flavor is not supported by this download server.`,
-      );
-    }
-    // We don't validate version since getReleaseInfo already does that
-    if (
-      options.platform &&
-      options.arch &&
-      !releaseInfo.files.includes(`${options.platform}-${options.arch}`)
-    ) {
-      throw new Error(
-        `Platform ${options.platform} and architecture ${options.arch} is not supported by this download server.`,
-      );
-    }
-    if (releaseInfo.components.node !== version.slice(1)) {
-      return new Error(
-        `NW.js ${releaseInfo.version} requires Node.js v${releaseInfo.components.node} but you are using Node.js ${version}.`,
-      );
-    }
-  } else if (options.mode === "run" || options.mode === "build") {
-    if (options.srcDir === "") {
-      throw new Error("srcDir is empty");
-    }
-    if (!releaseInfo.files.includes(`${options.platform}-${options.arch}`)) {
-      throw new Error(
-        `Platform ${options.platform} and architecture ${options.arch} is not supported by this download server. Sorry!`,
-      );
-    }
-    if (options.outDir) {
-      await readdir(options.outDir);
-    }
-    if (releaseInfo.components.node !== version.slice(1)) {
-      return new Error(
-        `NW.js ${releaseInfo.version} requires Node.js v${
-          releaseInfo.components.node
-        } but you are using Node.js ${version.slice(1)}.`,
-      );
-    }
-  } else {
+  if (releaseInfo.components.node > version.slice(1)) {
```

Comment:
I still recommend removing this check altogether. Most users really don't need it. If you keep it, change it to a warning instead of a critical error please.

Reply:
After giving it some thought I can see why warning is better. I was thinking about Native Node modules but there are a lot of apps which don't use it (also there's isn't a good way to rebuild native node modules).